src/releasenotes/notes/fix_parallel_diag_fusion-a7f914b3a9f491f7.yaml

```diff
@@ -0,0 +1,10 @@
+---
+fixes:
+  - |
+    If a number of gates exceeds ``fusion_parallelization_threshold``,
+    fusion transpiler divides the sequence of gates into sub-sequences
+    and multiple threads run transpilation for each sub-sequence.
+    A thread can fuse gates only in a sub-sequence. However, current
+    diagonal-fusion searches gates across sub-sequences and can produce
+    wrong results. This commit prevents from a thread not to fuse gates
+    across sub-sequences.
```

Comment:
The release note should be an explanation to the user, rather than to other developers - I think this is the text from the PR/commit body, but for users, we want to say something (very roughly) along the lines of

> Multi-threaded transpilations with a ``fusion_parallelization_threshold`` will now work correctly if the circuit can be fused with the diagonal strategy.  Previously, different threads would occasionally fuse the same element into multiple blocks, causing incorrect results.

(I don't 100% understand what `DiagonalFusion` is doing so feel free to correct it, but we don't need to be _super_ detailed about what caused a bug if the details weren't visible to the user, other than just giving incorrect results.)